feat: add new field size and update instance info layout

ajaypratap003 opened the pull request:
https://issues.redhat.com/browse/MGDSTRM-7507

Below changes covered in this PR

1.     Add new field `Size` on create modal. 

2.     Update Instance information section on create modal

3.     Update alert message for trial users 

4.     Add/update translation strings for trial and standard kafka and add help text for region field.

Note: these changes will be showing on storybook, not in production. 

`All updated stories are under KAFKA/Experimental `

jgiardino commented:
Related to this PR, I spent some time looking at the Slider component, and realized that it doesn't currently support linking the messages we want to show with the slider for accessibility via `aria-describedby`. I've raised an issue in PF for that here: https://github.com/patternfly/patternfly-react/issues/7164 

I don't see this as a blocker for this PR, but maybe it's worth adding the `aria-describedby` attributes now so that they automatically work as expected when we get a newer version of PF that adds this support, without requiring further updates on our end.

jgiardino commented:
@ajaypratap003 - It looks like this PR isn't ready for my review yet. When I open the storybook preview and navigate to the Create Kafka Instance stories, I don't see any updates. Please let me know if I'm looking in the wrong place. Otherwise, please ping me when you are ready for me to review updates in this PR.

ajaypratap003 commented:
> @ajaypratap003 - It looks like this PR isn't ready for my review yet. When I open the storybook preview and navigate to the Create Kafka Instance stories, I don't see any updates. Please let me know if I'm looking in the wrong place. Otherwise, please ping me when you are ready for me to review updates in this PR.

@jgiardino I think you are looking at wrong place. All the updates are under `KAKA/Experimental/Create Kafka Instance`.  Let me know still any issue to find stories.

jgiardino commented on locales/en/experimental/create-kafka-instance.json:
```diff
@@ -0,0 +1,106 @@
+  "size_field_aria": "More info for Size field",
+  "size_help_content": "Size of a Kafka instance is based on streaming units. The number of streaming units defines the capacity, resources, and limits of an instance. An instance with a larger size can handle higher loads and process larger amounts of events, has more storage, and can handle more clients and connections.",
+  "standard_kafka_size_dscription": "Learn how to add streaming units to your account",
+  "standard_kafka_streaming_units": "Your organization has {{streamingUnits}} streaming units remaining",
```
If the organization only has 1 streaming unit, how do we handle singular vs plural for translation? Do we have an option to use a string like "Your organization has 1 streaming unit remaining."? (i.e. without the "s" in "units")

jgiardino commented on locales/en/experimental/create-kafka-instance.json:
```diff
@@ -0,0 +1,106 @@
+  "streaming_size_value": "{{value}} streaming unit",
+  "all_region_unavailable_helper_text": "All regions in the selected cloud provider are temporarily unavailable. Select another cloud provider, try again later, or <0>contact support</0>.",
+  "some_region_unavailable_helper_text": "One or more regions in the selected cloud provider are temporarily unavailable. Select an available region, try again later, or <0>contact support</0>.",
+  "tria_all_region_unavailable_helper_text": "All regions in the selected cloud provider are temporarily unavailable. Select another cloud provider or try again later.",
```
minor typo in the text string id: `tria_...`

jgiardino commented:
Thanks for working on these updates! Overall, these updates look really good. Below is feedback based on review and/or discussion. I've grouped my feedback based on 1) specific components, 2) missing use cases, or 3) specific stories/use cases. Some of this feedback is not specific to this epic, but I'm including it here anyway. If there's an update that should be addressed as a separate update, then please let me know. 

**A couple of notes:**
- There is still a task open for Katie, our content designer, to review the contents and provide any updates. Hopefully this is just replacing existing text strings with updated strings. I don't expect this to be completed soon, so I don't want to block this PR for that, but wanted to let you know this was a pending update
- I still don't have an answer regarding the questions I've raised regarding messages with contact support links. I will try to get some resolution to this question this week. 

### 1) Specific components

**Footer message** - "Your Kafka instance will be ready for use shortly after creation."
After seeing all the examples of the modal, it seems really odd that this message displays when the Create instance button is disabled. I'd like to suggest that this message only displays when the Create instance button is enabled.

**Size input**

1. When a user has a subscription, the link “Learn how to add streaming units to your account” should not always display. It should only display if the helper text "Your organization has {x} streaming units remaining" displays as error text. This happens when a value selected in the Size input is larger than {x}. You can see examples of the UI [with the link](https://docs.google.com/presentation/d/1U4j27K3L-9Ry0GPe-fcKoU48dCPlEJsfoXJUpldAY3U/edit?pli=1#slide=id.g118b16caf36_0_93) and [without the link](https://docs.google.com/presentation/d/1U4j27K3L-9Ry0GPe-fcKoU48dCPlEJsfoXJUpldAY3U/edit?pli=1#slide=id.g1182510e32c_1_10) in the design doc. 
2. Should we show this behavior within the stories, or is it better to show these variations of the component by creating a component?
3. The "Size" input label should be capitalized.

**Details**
I reviewed some of these examples with other designers today. One point of feedback provided was that we should not display the Details section with a skeleton if we don't expect to ever display contents there. It's really only applicable in the loading state, or if we're waiting for information from the API after the user makes changes in the form. The reason for this is because the skeleton tells the user to wait for something to happen, but there are places where we're using the skeleton where nothing would actually ever display. The following examples are currently showing the skeleton for Details:

1. [Creation In Progress](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--creation-in-progress) - Can we display the size that the user had selected and show the details for that size in the Details section instead?
2. [No Regions Returned From Api For All Provider On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-for-all-provider-on-form-load) - For a subscription user, we can show the Size as disabled with the default value of 1 selected. This lets us show details for a size 1 Kafka instance instead of the skeleton. If the API doesn't support this use case, let me know and I can take this back to PF for feedback. 
3. [Over Quota On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--over-quota-on-form-load) - same comment as above
4. [Trial Unavailable On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-unavailable-on-form-load) - The size input displays as expected, but the Details section should show details for a trial instance
5. [Trial Used On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-used-on-form-load) - same comment as above



### 2) Missing use cases
The following is a use case outlined in the [messaging document](https://docs.google.com/document/d/1xaPia1F7QePDGoScdcpTwwgKHO_oU8L0D_cMNW_I1XA/edit#) that I did not see covered in the storybook preview: 

1. message 2.1.2 - The selected cloud provider has some regions that are unavailable. - You can see an example of this message in the [design document here](https://docs.google.com/presentation/d/1U4j27K3L-9Ry0GPe-fcKoU48dCPlEJsfoXJUpldAY3U/edit?pli=1#slide=id.g118b16caf36_0_129), although the message that's provided in the messaging document is different than what is shown in the example. I do see this use case represented for the trial user, but not for the user with a subscription.
2. message 2.1.3 - The selected region is not available for the selected size - You can see an example of this message in the [design document here](https://docs.google.com/presentation/d/1U4j27K3L-9Ry0GPe-fcKoU48dCPlEJsfoXJUpldAY3U/edit?pli=1#slide=id.g118b16caf36_0_83).


### 3) Specific stories / use cases

1. Loading data - We don’t know yet what quota exists for the user. Therefore helper text should not display for the Size slider.
3. [No Regions Returned From Api For All Provider On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-for-all-provider-on-form-load) - This example displays as expected, but it might be a duplicate of these other stories. Should any of these be removed, or are the covering unique cases? 
    * [All Regions Disabled On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--all-regions-disabled-on-form-load)
    * [No Regions Returned From Api On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-on-form-load)
4. [Over Quota On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--over-quota-on-form-load) - The helper text that displays for the Size input doesn't match the message at the top. Can you update the helper text here to say “Your organization has no streaming units remaining."
5. [Over Quota On Form Submit](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--over-quota-on-form-submit) - Message 3.1.1 should display above the form for this use case. The helper text that displays for the Size input should also be updated to match this use case. For example, if the user selected size 2, hits submit, and the quota dropped from 2 streaming units to 1 while the modal was open, then the form should display message 3.1.1 at the top, Size should still show the value "2" as selected, and this error text would display below Size: "Your organization has 1 streaming unit remaining."
6. [Trial All Regions Disabled On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-all-regions-disabled-on-form-load) - The message that displays here is correct, except the message should only display for one of the cloud providers, similar to the behavior in [No Regions Returned From Api For A Provider On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-for-a-provider-on-form-load). For this story, should the name be updated to be consistent with the non-trial version, e.g. "Trial No Regions Returned From Api For A Provider On Form Load"
7. [Trial Available On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-available-on-form-load) - Use message 1.2.5 as the helper text for Cloud region instead
8. [Trial Selected Region Unavailable On Form Submit](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-selected-region-unavailable-on-form-submit) - I originally proposed using  message 3.2.1 instead of "Address form errors to proceed". But if there are other form errors, then this results in too many messages at the top of the form. Another idea is to use message 2.2.2 but using the error variant. Let me know what you think about this idea.
10. [Trial Unavailable On Form Submit](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-unavailable-on-form-submit) - I didn't provide an update for this story, and it's different from what's there today. But based on other changes we're making for [Trial Unavailable On Form Load](https://redhat-developer.github.io/app-services-ui-components/?path=/story/kafka-create-kafka-instance-create-kafka-instance--trial-unavailable-on-form-load), I added an updated message for this story too: message 3.2.2. This use case should also use message 1.2.5 for the cloud region input helper text.
11. [Trial Unavailable On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-unavailable-on-form-load) - This use case should use message 1.2.5 for the cloud region input helper text
12. [Trial Used On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-used-on-form-load) - Based on recent discussion with the BU, I provided an updated message for this use case under message 1.2.3

ajaypratap003 commented:
> Thanks for working on these updates! Overall, these updates look really good. Below is feedback based on review and/or discussion. I've grouped my feedback based on 1) specific components, 2) missing use cases, or 3) specific stories/use cases. Some of this feedback is not specific to this epic, but I'm including it here anyway. If there's an update that should be addressed as a separate update, then please let me know.
> 
> **A couple of notes:**
> 
> * There is still a task open for Katie, our content designer, to review the contents and provide any updates. Hopefully this is just replacing existing text strings with updated strings. I don't expect this to be completed soon, so I don't want to block this PR for that, but wanted to let you know this was a pending update
> * I still don't have an answer regarding the questions I've raised regarding messages with contact support links. I will try to get some resolution to this question this week.
> 
> ### 1) Specific components
> **Footer message** - "Your Kafka instance will be ready for use shortly after creation." After seeing all the examples of the modal, it seems really odd that this message displays when the Create instance button is disabled. I'd like to suggest that this message only displays when the Create instance button is enabled.
> 
Yes, we can add condition for showing alert message when Create button will be enabled but there are many cases when create button is disabled, specially when user click on create button and button is disabled during creating kafka and  message will be hide suddenly.  If you are okay with message hide during creating kafka then I can add check.

> **Size input**
> 
> 1. When a user has a subscription, the link “Learn how to add streaming units to your account” should not always display. It should only display if the helper text "Your organization has {x} streaming units remaining" displays as error text. This happens when a value selected in the Size input is larger than {x}. You can see examples of the UI [with the link](https://docs.google.com/presentation/d/1U4j27K3L-9Ry0GPe-fcKoU48dCPlEJsfoXJUpldAY3U/edit?pli=1#slide=id.g118b16caf36_0_93) and [without the link](https://docs.google.com/presentation/d/1U4j27K3L-9Ry0GPe-fcKoU48dCPlEJsfoXJUpldAY3U/edit?pli=1#slide=id.g1182510e32c_1_10) in the design doc.
> 2. Should we show this behavior within the stories, or is it better to show these variations of the component by creating a component?
> 3. The "Size" input label should be capitalized.
>
> **Details** I reviewed some of these examples with other designers today. One point of feedback provided was that we should not display the Details section with a skeleton if we don't expect to ever display contents there. It's really only applicable in the loading state, or if we're waiting for information from the API after the user makes changes in the form. The reason for this is because the skeleton tells the user to wait for something to happen, but there are places where we're using the skeleton where nothing would actually ever display. The following examples are currently showing the skeleton for Details:
> 
> 1. [Creation In Progress](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--creation-in-progress) - Can we display the size that the user had selected and show the details for that size in the Details section instead?

agreed, we should display size details. Already showing user selected size  in details section. 

> 2. [No Regions Returned From Api For All Provider On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-for-all-provider-on-form-load) - For a subscription user, we can show the Size as disabled with the default value of 1 selected. This lets us show details for a size 1 Kafka instance instead of the skeleton. If the API doesn't support this use case, let me know and I can take this back to PF for feedback.

API doesn't support this case.  API needs cloud provider and cloud region for getting available sizes. In this case cloud region is not available.

> 3. [Over Quota On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--over-quota-on-form-load) - same comment as above

API doesn't support this case,  API needs cloud provider and cloud region which are not available in this case.

> 4. [Trial Unavailable On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-unavailable-on-form-load) - The size input displays as expected, but the Details section should show details for a trial instance

Same as mentioned above. 
API doesn't support this case,  API needs cloud provider and cloud region which are not available in this case.

> 5. [Trial Used On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-used-on-form-load) - same comment as above
> 

Same as mentioned above. 
API doesn't support this case,  API needs cloud provider and cloud region which are not available in this case.

As we don't have instance details in many cases but we can display message in details section instead of skeleton.


> ### 2) Missing use cases
> The following is a use case outlined in the [messaging document](https://docs.google.com/document/d/1xaPia1F7QePDGoScdcpTwwgKHO_oU8L0D_cMNW_I1XA/edit#) that I did not see covered in the storybook preview:
>
> 1. message 2.1.2 - The selected cloud provider has some regions that are unavailable. - You can see an example of this message in the [design document here](https://docs.google.com/presentation/d/1U4j27K3L-9Ry0GPe-fcKoU48dCPlEJsfoXJUpldAY3U/edit?pli=1#slide=id.g118b16caf36_0_129), although the message that's provided in the messaging document is different than what is shown in the example. I do see this use case represented for the trial user, but not for the user with a subscription.

Not sure but I can see both case.
Standard kafka
![image](https://user-images.githubusercontent.com/11775081/163030706-6bc93e1f-746f-4a3d-9a2e-9935b3bc73f0.png)


Trial kafka
![image](https://user-images.githubusercontent.com/11775081/163030773-2abc1c23-7950-4f61-9eb3-bf61b2bd67fa.png)


> 2. message 2.1.3 - The selected region is not available for the selected size - You can see an example of this message in the [design document here](https://docs.google.com/presentation/d/1U4j27K3L-9Ry0GPe-fcKoU48dCPlEJsfoXJUpldAY3U/edit?pli=1#slide=id.g118b16caf36_0_83).
> 

Pending. Needs to cover this use case which is depends on API data. working on it.

> ### 3) Specific stories / use cases
> 1. Loading data - We don’t know yet what quota exists for the user. Therefore helper text should not display for the Size slider.

Noted 

> 2. [No Regions Returned From Api For All Provider On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-for-all-provider-on-form-load) - This example displays as expected, but it might be a duplicate of these other stories. Should any of these be removed, or are the covering unique cases?
>    
>    * [All Regions Disabled On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--all-regions-disabled-on-form-load)
>    * [No Regions Returned From Api On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-on-form-load)

Both are different use cases. 
In first case, API is returning empty list of data.
In second case API is returning data but unsupported regions which are display in list as disabled. 

> 3. [Over Quota On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--over-quota-on-form-load) - The helper text that displays for the Size input doesn't match the message at the top. Can you update the helper text here to say “Your organization has no streaming units remaining."

Noted

> 4. [Over Quota On Form Submit](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--over-quota-on-form-submit) - Message 3.1.1 should display above the form for this use case. The helper text that displays for the Size input should also be updated to match this use case. For example, if the user selected size 2, hits submit, and the quota dropped from 2 streaming units to 1 while the modal was open, then the form should display message 3.1.1 at the top, Size should still show the value "2" as selected, and this error text would display below Size: "Your organization has 1 streaming unit remaining."

This case is pending. I've added for on load form. 
All the error are showing under form on form submit. 


> 5. [Trial All Regions Disabled On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-all-regions-disabled-on-form-load) - The message that displays here is correct, except the message should only display for one of the cloud providers, similar to the behavior in [No Regions Returned From Api For A Provider On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-for-a-provider-on-form-load). For this story, should the name be updated to be consistent with the non-trial version, e.g. "Trial No Regions Returned From Api For A Provider On Form Load"

This use case similar to  [all-regions-disabled-on-form-load(]https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--all-regions-disabled-on-form-load). 
This use case for trial and have trial alert message. 

> 6. [Trial Available On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-available-on-form-load) - Use message 1.2.5 as the helper text for Cloud region instead

Noted

> 7. [Trial Selected Region Unavailable On Form Submit](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-selected-region-unavailable-on-form-submit) - I originally proposed using  message 3.2.1 instead of "Address form errors to proceed". But if there are other form errors, then this results in too many messages at the top of the form. Another idea is to use message 2.2.2 but using the error variant. Let me know what you think about this idea.

Using message 3.2.1 as suggested in doc. Here we have 2 messages one for trial info and second for error on submit.
This use case seems fine. 

> 8. [Trial Unavailable On Form Submit](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-unavailable-on-form-submit) - I didn't provide an update for this story, and it's different from what's there today. But based on other changes we're making for [Trial Unavailable On Form Load](https://redhat-developer.github.io/app-services-ui-components/?path=/story/kafka-create-kafka-instance-create-kafka-instance--trial-unavailable-on-form-load), I added an updated message for this story too: message 3.2.2. This use case should also use message 1.2.5 for the cloud region input helper text.

Noted. will update message 1.2.5

> 9. [Trial Unavailable On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-unavailable-on-form-load) - This use case should use message 1.2.5 for the cloud region input helper text

Noted. will update message 1.2.5

> 10. [Trial Used On Form Load](https://619e1921b3e38b003afeb2be-iuccxxzxdr.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-used-on-form-load) - Based on recent discussion with the BU, I provided an updated message for this use case under message 1.2.3

Noted


@jgiardino Thanks for adding all these feedback. I'll update all the translation strings. 
Some use case are pending based on size selection. I'm working on that. Let's review together all these changes tomorrow.

jgiardino commented:
> API doesn't support this case. API needs cloud provider and cloud region for getting available sizes. In this case cloud region is not available.

During a design review earlier today, we were discussing the contents of [this Overview page](https://marvelapp.com/prototype/2eb7b0hg/screen/86196360) where we do want to show all the sizes. According to PM, the api calls for getting cloud providers and getting limits per size should not be dependent on each other. There wasn't anyone from the backend team to confirm this with. 

Here is a sketch that @wtrocki provided about API calls. Not sure if this answers this question, so maybe this would be something to follow-up on.

wtrocki commented:
<img width="1395" alt="Screenshot 2022-03-24 at 14 56 23" src="https://user-images.githubusercontent.com/981838/163219993-3e4e7d66-be53-4989-8534-539802371bbc.png">

jgiardino commented:
@ajaypratap003 - I shared this link that you provided with @DuncanDoyle: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/843

This was his response:
> That endpoint: yes. It's needed for things like fleet isolation. But there has also been a discussion about an endpoint that simply returns what is generally supported by the service. And I'm not sure if and how that is implemented.

Not sure if @wtrocki or @riccardo-forina might have more insights into this or not. 

But if we're depending on providing a cloud provider and region to return a list of sizes, this means:

1. We can't show contents in the Details section for the default size of 1 SU when there's an issue with region availability. This is an unlikely edge case, so maybe it's fine to just display the skeleton here. 
2. Every time the user makes a cloud provider or region selection, will we have to make a new API call to load details about the size? That seems a bit excessive.

jgiardino commented:
@ajaypratap003 - Thanks for your comments and for walking through everything with me today. Here's a recap of what we discussed for everyone's awareness...

**Re: my comment under 1) Specific components > Footer message,** we discussed handling this as a future update. It's nice-to-have, but not critical for this work. 

**In my original comment under 1) Specific components > Details,** where I noted that the Details skeleton shouldn't display here for the use cases listed — we discussed that this is not a blocker for this PR. It would be good to confirm whether the API does or does not support this case and revisit this issue. I also followed up with PF design regarding an alternate version of the skeleton we could use here, or some of the recommendation. 

**You raised a question about how to map size id to slider values,** since the size id is not numeric, and slider values expect numbers. This is outside my area of expertise 😉 Maybe @riccardo-forina has suggestions here. 

**You raised a question about what happens if the user clears the cloud region selection.** The expectation is that a default is selected for both cloud provider and cloud region, and the option to clear a selection in the menu is not provided. 

I think @riccardo-forina's suggestion was to just default to the first option returned by the API. However, there is a possibility that the first region is unavailable, as shown in [Trial Some Regions Disabled On Form Load](https://619e1921b3e38b003afeb2be-gqirknoxow.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-some-regions-disabled-on-form-load). I think for this case, we would default to the first enabled option. 

**But then you raised a question about how to display the cloud region menu when none of the regions are available,** and I needed to think about this one.

Below are screencaps of your current implementation in [All Regions Disabled On Form Load](https://619e1921b3e38b003afeb2be-gqirknoxow.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--all-regions-disabled-on-form-load) vs what I would recommend instead.

| current implementation | recommended implementation |
| -- | -- | 
| <img width="534" alt="image" src="https://user-images.githubusercontent.com/21063328/163279703-ff425ca8-2347-495a-8ccc-8ec04a155758.png"> | <img width="529" alt="image" src="https://user-images.githubusercontent.com/21063328/163280708-d3807943-256c-41a6-9056-4eb7ac02450a.png"> |
| You currently show a placeholder "Select region" | If all regions in the menu are disabled or no regions are returned by the API, the placeholder to "Regions temporarily unavailable" |
| <img width="529" alt="image" src="https://user-images.githubusercontent.com/21063328/163280940-8e502dba-67df-4e4b-a030-3c1c5758d351.png"> | <img width="534" alt="image" src="https://user-images.githubusercontent.com/21063328/163280795-3f43cbba-bbc2-4b19-85f3-a3ab0f516055.png"> |
| Current implementation is to include the placeholder text in the menu | All implementations of the Cloud region menu should display the options only. For the use case where all regions in the menu are disabled, the user can see the regions, but can't select any. For the use case where the API isn't providing regions, we can just disable the select component altogether, but keep the placeholder text as "Regions temporarily unavailable" |

**You also asked if form inputs should be disabled in [All Regions Disabled On Form Load](https://619e1921b3e38b003afeb2be-gqirknoxow.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--all-regions-disabled-on-form-load) like they are in [No Regions Returned From Api For All Provider On Form Load](https://619e1921b3e38b003afeb2be-gqirknoxow.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-for-all-provider-on-form-load).** I'd like to double-check this question with others. I'm on the side of not disabling form inputs for these cases. I think there's value in seeing what options are available if we have options to show, even though these options aren't currently available. I do think there are cases where it makes sense to not have disabled form inputs, though, like the creation in progress or error on form load. 

**We discussed the error shown in [Over Quota On Form Submit](https://619e1921b3e38b003afeb2be-gqirknoxow.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--over-quota-on-form-submit),** and noted that the implementation doesn't match the messaging doc regarding alert variant. I updated the doc to match your implementation.

I think that covers everything, but please let me know if there's any other information you need from me on this.

jgiardino commented on locales/en/experimental/create-kafka-instance.json:
```diff
@@ -0,0 +1,107 @@
+  "streaming_unit": "streaming units",
+  "trial_kafka_size_description": "With a subscription, you can select a size for your instance. Changing the size allows you to increase limits such as ingress, egress, and storage. The limits that you get with a trial instance are listed under Details.",
+  "streaming_size_value": "{{value}} streaming unit",
+  "all_region_unavailable_helper_text": "All regions in the selected cloud provider are temporarily unavailable. Select another cloud provider, try again later.",
```
Replace this:
> Select another cloud provider, try again later.
With this:
> Select another cloud provider or try again later.

jgiardino commented on locales/en/experimental/create-kafka-instance.json:
```diff
@@ -0,0 +1,107 @@
+  "trial_kafka_size_description": "With a subscription, you can select a size for your instance. Changing the size allows you to increase limits such as ingress, egress, and storage. The limits that you get with a trial instance are listed under Details.",
+  "streaming_size_value": "{{value}} streaming unit",
+  "all_region_unavailable_helper_text": "All regions in the selected cloud provider are temporarily unavailable. Select another cloud provider, try again later.",
+  "some_region_unavailable_helper_text": "One or more regions in the selected cloud provider are temporarily unavailable. Select an available region, try again later.",
```
Replace this:
> Select an available region, try again later.

With this:
> Select an available region or try again later.

jgiardino commented:
For this last review, I focused mainly on the messages that are displayed in each example. The following are stories where the messages aren't displaying as expected. There are likely other refinements that are needed (some are noted below as "nice to have"), but I consider the messages the highest priority to address. We can discuss additional refinements as part of a follow-up PR.

1. [All Regions Disabled On Form Load](https://619e1921b3e38b003afeb2be-rvbzryrfld.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--all-regions-disabled-on-form-load)
    1. inline alert above form looks good
    2. error helper text for Cloud region should not display in this example - this message is only for the use case when at least one provider has at least 1 region available

2. [No Regions Returned From Api For A Provider On Form Load](https://619e1921b3e38b003afeb2be-rvbzryrfld.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-for-a-provider-on-form-load)
    1. inline alert above form should not display
    1. Nice to have - Cloud region should be disabled in this example when AWS is selected, since the menu is empty

3. [No Regions Returned From Api For All Provider On Form Load](https://619e1921b3e38b003afeb2be-rvbzryrfld.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-for-all-provider-on-form-load)
    1. Nice to have - Show 1 streaming unit as the default size that’s selected

4. [No Regions Returned From Api On Form Load](https://619e1921b3e38b003afeb2be-rvbzryrfld.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--no-regions-returned-from-api-on-form-load)
    1. This appears to be a duplicate of the previous one. The messaging in the previous one matches what I expect to see when there are no regions for any cloud provider. Should we just remove this story?

5. [Some Regions Disabled On Form Load](https://619e1921b3e38b003afeb2be-rvbzryrfld.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--some-regions-disabled-on-form-load)
    1. helper text messages (aside from typos noted inline) look good
    1. inline alert above form should not display when Azure is selected. This message is only intended for the use case when all regions (in all providers) are unavailable.

6. [Trial All Regions Disabled On Form Load](https://619e1921b3e38b003afeb2be-rvbzryrfld.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-all-regions-disabled-on-form-load)
    1. For this case, we should see the same inline alert that displays in this example: [Trial Unavailable On Form Load](https://619e1921b3e38b003afeb2be-rvbzryrfld.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-unavailable-on-form-load). If regions are unavailable, then trials are unavailable, and that’s the primary message we want to display in this case. 
    1. We also would not show the error text for cloud region. We don’t want to tell them to select another cloud provider when none of the cloud providers have any available regions. 

7. [Trial Unavailable On Form Submit](https://619e1921b3e38b003afeb2be-rvbzryrfld.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-create-kafka-instance--trial-unavailable-on-form-submit)
    1. This story should have no changes in this PR (except for the updated info message), but I see that it’s different from what we have now: https://619e1921b3e38b003afeb2be-lxiyfbqvig.chromatic.com/?path=/story/kafka-create-kafka-instance-create-kafka-instance--trial-unavailable-on-form-submit

ajaypratap003 commented:
@jgiardino I'll be on PTO for one week. @riccardo-forina will take care about this PR and large kafka work.

jgiardino commented on locales/en/experimental/create-kafka-instance.json:
```diff
@@ -0,0 +1,111 @@
+  "standard_kafka_streaming_unit_other": "Your organization has {{count}} streaming units remaining",
+  "select_cloud_provider": "Select cloud provider",
+  "select_region": "Select region",
+  "sizes_missing": "Select a cloud provider and region to find the available sizes",
```
I think it's rare that this text will display, assuming we will always select default values. But in some of the examples, this message displays when the form is disabled. For now, let's use this message so that we still communicate that the size input is dependent on cloud provider and region selection, without telling them to make a selection they can't make:
> Size options display when a cloud provider and region are selected

jgiardino commented on src/Kafka/Experimental/CreateKafkaInstance/components/FieldCloudRegion.tsx:
```diff
@@ -0,0 +1,68 @@
+            <HelperTextItem variant="warning" hasIcon>
+              {t("some_region_unavailable_helper_text")}
+            </HelperTextItem>
+          </HelperText>
```
After we reviewed this together, I realized that helper text provided by the form group component includes xs margin-top. But when adding the helper text using the helper text component, those styles aren't getting applied. 

You could include the class `pf-c-form__helper-text` on the `<HelperText>` component to provide additional margin-top. Or the utility class would work, too (until they update the form group component to use a different size): `pf-u-mt-xs`

jgiardino commented on src/Kafka/CreateKafkaInstance/components/TrialKafkaForm.tsx:
```diff
@@ -0,0 +1,271 @@
+          return (
+            <div className="pf-c-form__helper-text pf-m-warning">
+              {t("trial_some_region_unavailable_helper_text")}
+            </div>
```
The  FieldCloudRegion component was updated to use the HelperText component instead. Do the same updates need to be made here?

jgiardino commented on src/Kafka/Experimental/CreateKafkaInstance/components/FieldSize.tsx:
```diff
@@ -0,0 +1,152 @@
+          {t("standard_kafka_size_description")}
+        </Button>
+      </HelperTextItem>
+    </HelperText>
```
The button that's included here should be included in a separate `<HelperTextItem>` component. For example:
```
    <HelperText>
      <HelperTextItem variant="error" hasIcon>
        <div>
          {t("standard_kafka_streaming_unit", {
            count: remainingStreamingUnits,
          })}
        </div>
      </HelperTextItem>
      <HelperTextItem>
        <Button
          className="pf-c-form__helper-text"
          variant={ButtonVariant.link}
          isInline
          onClick={onLearnHowToAddStreamingUnits}
        >
          {t("standard_kafka_size_description")}
        </Button>
      </HelperTextItem>
    </HelperText>
```

This should result in the button being left-aligned with the icon, as shown below:
![image](https://user-images.githubusercontent.com/21063328/165838272-b531da97-09a3-4ed0-90aa-1ff91e1567fc.png)

jgiardino commented on src/Kafka/Experimental/CreateKafkaInstance/components/FieldSize.tsx:
```diff
@@ -0,0 +1,152 @@
+  );
+
+  const helperTextOverQuota = (
+    <HelperText>
```
See my [other comment](https://github.com/redhat-developer/app-services-ui-components/pull/252/files#r861273813) about applying margin-top to the `<HelperText>` component.

jgiardino commented:
These updates look great! Thanks for walking me though this earlier today. 

I included some comments inline for specific components. Here are other items I noticed. There are some items that I noted as *minor* meaning I don't think it's a big deal to address them at this time. My main focus is that messaging is clear and accurate, and that we align the UI with stakeholder expectations. 

### For all of the trial examples
The [original design](https://docs.google.com/presentation/d/1U4j27K3L-9Ry0GPe-fcKoU48dCPlEJsfoXJUpldAY3U/edit?pli=1#slide=id.g118b16caf36_0_0) included a disabled slider with no value marked as selected (i.e. value is set to `-1`). The goal behind this is to show the slider along with specific messaging to communicate that sizes are available with a subscription. But we don’t want to display a value as selected, since these sizes are relevant to the standard type only. 

I also think we could display this form group exactly the same for all trial examples (e.g. [All regions Unavailable - Trial](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-load-create-kafka-instance--trial-all-regions-disabled-on-form-load), [Quota Available - Trial](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-load-create-kafka-instance--trial-available-on-form-load)). Even if the user can’t create a trial instance, we still want to communicate that sizes are available with a subscription. But I also realize that the API expects a cloud provider and region to get a set of sizes, and that it’s possible for the API to eventually return a different set of sizes for different cloud providers. Here are some suggestions for how to address this: 

1. Display a slider with sizes 1 and 2 for now, and we revisit this with the larger team. 
2. Keep the behavior as you have it (i.e. slider only displays if cloud provider and region are selected), but display the following as helper text instead of "Select a cloud provider and region to ...". With this update, we're always displaying the following message as helper text for the Size input when the user is getting a trial instance.
    > With a subscription, you can select a size for your instance. Changing the size allows you to increase limits such as ingress, egress, and storage. The limits that you get with a trial instance are listed under Details. 
    > [Learn more about sizes](https://console.redhat.com/application-services/streams/overview)"

### For specific examples

1. [All Regions Unavailable - Standard](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-load-create-kafka-instance--all-regions-disabled-on-form-load)
    - **minor** - don’t show a cloud provider as selected
    - **minor** - don’t show the error icon in the cloud region menu. These are usually accompanied by helper text messaging
2. [Quota Available - Standard](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-load-create-kafka-instance--quota-available-on-form-load)
   - When reviewing this together, we discussed having a fallback default value to select the first option in the list. I thought you made this update already, but I’m not seeing it in this example or other examples where Cloud region is enabled. 
3. [All regions Unavailable - Trial](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-load-create-kafka-instance--trial-all-regions-disabled-on-form-load)
    - A different message should display in this case: 
      > **Kafka instances are unavailable for creation**  
      > All available trial instances are currently in use. Try again later.
    - See note above about the size input. At a minimum, we should display the following as helper text instead:  
      > With a subscription, you can select a size for your instance. Changing the size allows you to increase limits such as ingress, egress, and storage. The limits that you get with a trial instance are listed under Details.  Learn more about sizes
    - **minor** - don’t show a cloud provider as selected
    - **minor** - don’t show the error icon in the cloud region menu. These are usually accompanied by helper text messaging
4. [Trial Unavailable - Trial](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-load-create-kafka-instance--trial-unavailable-on-form-load)
    - a warning message is missing: 
      > **Kafka instances are unavailable for creation**  
      > All available trial instances are currently in use. Try again later.
5. [Loading Data](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-processing-create-kafka-instance--loading-data)
    - The message that displays for the Size input seems out of place at this time. The original design showed the size slider as disabled without an option selected and without any helper text. If that isn’t a good solution, another idea is to just display a skeleton in place of any input or helper text. 
6. [Over Quota - Standard](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-submit-create-kafka-instance--over-quota-on-form-submit)
    - **Minor** - In this case, I think we’d want to keep the same value selected after form submit that the user specified for size. If I’m seeing a message that I only have 3 units remaining, then we should show 5 as selected. 
7. [Selected region unavailable - Standard](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-submit-create-kafka-instance--selected-region-unavailable-on-form-submit)
    - The inline error alert is slightly off. This should be the message that displays: 
      > **The selected cloud provider region is temporarily unavailable.** 
      > Select another region or try again later.
    - For this use case, the cloud region select menu would have at least one region that is disabled with the description “Temporarily unavailable”. It would be good to add that to the example.
    - The original helper text that I specified is the error and warning text that we display on form load when regions are not available. But those messages don’t really fit if we keep the originally selected value. I could be convinced either way (keep the selected value or clear it). I’m leaning toward clearing it for now and using one of the messages. But we could also keep the originally selected value and display this error helper text:
      - If all regions are disabled for the selected cloud provider, but another cloud provider has some available regions:
        >  All regions in the selected cloud provider are temporarily unavailable. Select another cloud provider or try again later.
      - If some regions are disabled: 
        > The selected cloud provider region is temporarily unavailable. Select another region or try again later.
      - If all regions in all providers are unavailable, this is likely a separate story. We would want to show a different message above the form:
        >  **Kafka instances are unavailable for creation** 
        > Cloud provider regions are temporarily unavailable. Try again later.  
8. [Selected region unavailable - Trial](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-submit-create-kafka-instance--trial-selected-region-unavailable-on-form-submit)
    - Same comments about helper text for 7. apply here, with this one exception:
      - If all regions in all providers are unavailable, this use case is covered by [Trial Unavailable - Trial](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-submit-create-kafka-instance--trial-unavailable-on-form-submit) 
    - A Required error message displays under the AZ toggle group, but we will always have a default selected. 
9. [Trial Unavailable - Trial](https://619e1921b3e38b003afeb2be-uprfxxbloo.chromatic.com/?path=/story/kafka-experimental-create-kafka-instance-stories-form-submit-create-kafka-instance--trial-unavailable-on-form-submit)
    - A Required error message displays under the AZ toggle group, but we will always have a default selected. 
    - The inline error alert is incorrect. This should be the message that displays: 
      > **Kafka instance cannot be created**
      > All available trial instances are currently in use. Try again later.

riccardo-forina commented:
Thank you for the review Jenn, it was extremely useful as always!
I did a lot of bugfixing and aim correction, hopefully it's better now. I also moved the component to a new place, I ditched the Experimental folder in favour of giving the component a new name; it's now called "Create Kafka Instance With Size". This gives us the option to support both creation flows in MAS UI, if we need to do that.

jgiardino commented on locales/en/create-kafka-instance-with-sizes.json:
```diff
@@ -0,0 +1,109 @@
+    "trial_available_title": "You can create a trial instance to evaluate this service.",
+    "trial_unavailable_message": "All available trial instances are currently in use. Try again later.",
+    "trial_unavailable_title": "$t(common:kafka) instances are unavailable for creation",
+    "trial_used_message": "You can deploy 1 instance at a time. To deploy a new instance, delete your existing one first. If you are interested in using this service in production, go to the <0>Streams for Apache Kafka Overview</0> to see what options are available during creation and learn how to contact us.",
```
Suggestion is to clarify that this is for trial instances. For example:
> You can deploy 1 trial instance at a time. To deploy a new instance,...

jgiardino commented on locales/en/create-kafka-instance-with-sizes.json:
```diff
@@ -0,0 +1,109 @@
+  "storage_value": "up to {{value}} GiB",
+  "stream_unit_value": "{{size}} streaming units",
+  "streaming_unit": "streaming units",
+  "trial_kafka_size_description": "With a subscription, you can select a size for your instance. Changing the size allows you to increase limits such as ingress, egress, and storage. The limits that you get with a trial instance are listed under Details.",
```
PM requested to make this shorter. All of the contents need a full review from our content designer, but for now, let's change this to the following:
> With a subscription, you can change the capacity of your instance by selecting a size. The capacity that you get with a trial instance is listed under Details.

jgiardino commented:
Your updates looks great! Here are some things I'm noticing in a follow-up review:

### [No regions returned from API for a provider - Standard](https://619e1921b3e38b003afeb2be-hrnjhwszdu.chromatic.com/?path=/story/kafka-create-kafka-instance-with-sizes-stories-form-load-create-kafka-instance--no-regions-returned-from-api-for-a-provider-on-form-load)
The following message displays. 
<img width="551" alt="image" src="https://user-images.githubusercontent.com/21063328/165970116-fd591864-5daa-4985-b8f8-e33ad1bb8219.png">
This message should only display if there are some regions unavailable in the menu. Instead, the following message should display as error helper text:
> All regions in the selected cloud provider are temporarily unavailable. Select another cloud provider or try again later.

### [All regions Unavailable - Trial](https://619e1921b3e38b003afeb2be-hrnjhwszdu.chromatic.com/?path=/story/kafka-create-kafka-instance-with-sizes-stories-form-load-create-kafka-instance--trial-all-regions-disabled-on-form-load)
Currently I see two messages displaying, but in this case, I should only see the warning message.
<img width="858" alt="image" src="https://user-images.githubusercontent.com/21063328/165976545-1f91a1dd-3655-435c-bf7c-2cfc8308e2e4.png">


### [Over Quota - Trial](https://619e1921b3e38b003afeb2be-hrnjhwszdu.chromatic.com/?path=/story/kafka-create-kafka-instance-with-sizes-stories-form-load-create-kafka-instance--trial-used-on-form-load)
Similar issue as previous section - I should only see the warning message here.
<img width="857" alt="image" src="https://user-images.githubusercontent.com/21063328/165976804-c4f82950-30d6-422a-9faf-2be44073e7ad.png">

### [Loading Data](https://619e1921b3e38b003afeb2be-hrnjhwszdu.chromatic.com/?path=/story/kafka-create-kafka-instance-with-sizes-stories-form-processing-create-kafka-instance--loading-data)
**Minor** - For the Size input, it would be ideal if we don't display any helper text and just display a skeleton instead, since we don't know yet if the user has a subscription or not.

<img width="531" alt="image" src="https://user-images.githubusercontent.com/21063328/165977046-4cd22aaa-7bd2-402a-bc4b-cce1e6704070.png">

### [Selected region unavailable - Standard](https://619e1921b3e38b003afeb2be-hrnjhwszdu.chromatic.com/?path=/story/kafka-create-kafka-instance-with-sizes-stories-form-submit-create-kafka-instance--selected-region-unavailable-on-form-submit) and [Selected region unavailable - Trial](https://619e1921b3e38b003afeb2be-hrnjhwszdu.chromatic.com/?path=/story/kafka-create-kafka-instance-with-sizes-stories-form-submit-create-kafka-instance--trial-selected-region-unavailable-on-form-submit)
The title in this alert should be the following instead: 
> The selected cloud provider region is temporarily unavailable. 

And this text would be the body:
> Select another region or try again later.

<img width="861" alt="image" src="https://user-images.githubusercontent.com/21063328/165977234-8d5af27d-5e95-4f25-8b55-d3ad41697d90.png">

jgiardino commented on locales/en/create-kafka-instance-with-sizes.json:
```diff
@@ -0,0 +1,110 @@
+  "streaming_unit": "streaming units",
+  "temporarily_unavailable": "Temporarily unavailable",
+  "trial_all_region_unavailable_helper_text": "All regions in the selected cloud provider are temporarily unavailable. Select another cloud provider or try again later.",
+  "trial_kafka_size_description": "With a subscription, you can change the capacity of your instance by selecting a size. The capacity that you get with a trial instance is listed under Details.",
```
Sorry for the extra update on this string. Katie was having laptop issues today, and I wanted to share her suggestion since it's much better than mine:  

> Size selection is available with a subscription. Trial instances have limited capacity, and their limits are listed under Details.

jgiardino commented on locales/en/create-kafka-instance-with-sizes.json:
```diff
@@ -0,0 +1,110 @@
+{
+  "all_region_unavailable_helper_text": "All regions in the selected cloud provider are temporarily unavailable. Select another cloud provider or try again later.",
```
The original message I provided was too far out in the future, when we have more than one cloud provider. Until we have support for another cloud provider, we shouldn't tell the user to select another cloud provider. But there might also be future situations where we support more than one cloud provider, but don't provide an option.  So maybe we keep this and add another message, or maybe we just update this message for now.

In the case where only one cloud provider is supported, we should say this:
> All regions in the selected cloud provider are temporarily unavailable. Try again later.

jgiardino replied:
I'm taking back my suggestion for this message.

This helper text noted in my previous comment should only display for a use case where 2 or more providers are listed, and the selected provider has no available regions, but at least one region is available in another provider.

We should display the following message if there are no available regions for any provider. This message works whether we display one or more providers.
<img width="851" alt="image" src="https://user-images.githubusercontent.com/21063328/166305120-7b93f6cd-10f1-4a0b-aa6c-9b09e344783a.png">

jgiardino commented on locales/en/create-kafka-instance-with-sizes.json:
```diff
@@ -0,0 +1,110 @@
+  "streaming_size_value": "{{value}} streaming unit",
+  "streaming_unit": "streaming units",
+  "temporarily_unavailable": "Temporarily unavailable",
+  "trial_all_region_unavailable_helper_text": "All regions in the selected cloud provider are temporarily unavailable. Select another cloud provider or try again later.",
```
Same comment as above...
The original message I provided was too far out in the future, when we have more than one cloud provider. Until we have support for another cloud provider, we shouldn't tell the user to select another cloud provider. But there might also be future situations where we support more than one cloud provider, but don't provide an option.  So maybe we keep this and add another message, or maybe we just update this message for now.

In the case where only one cloud provider is supported, we should say this:
> All regions in the selected cloud provider are temporarily unavailable. Try again later.

jgiardino commented on src/Kafka/CreateKafkaInstanceWithSizes/components/FieldCloudRegion.tsx:
```diff
@@ -0,0 +1,86 @@
+            <HelperTextItem variant="warning" hasIcon>
+              {someRegionsUnavailable
+                ? t("some_region_unavailable_helper_text")
+                : t("all_region_unavailable_helper_text")}
```
When I look at [No regions returned from API for a provider - Standard](https://619e1921b3e38b003afeb2be-tvvapjvikl.chromatic.com/?path=/story/kafka-create-kafka-instance-with-sizes-stories-form-load--no-regions-returned-from-api-for-a-provider-on-form-load), I see this helper text display with the warning variant. 

But I expect this helper text to display using the error variant. Only the  `some_region_unavailable_helper_text` helper text should use the warning variant.

<img width="558" alt="image" src="https://user-images.githubusercontent.com/21063328/166305905-99cacc25-c171-4f38-8fdd-3a5adadb2f04.png">

jgiardino commented on src/Kafka/CreateKafkaInstanceWithSizes/components/FieldCloudRegion.tsx:
```diff
@@ -0,0 +1,86 @@
+      helperText={
+        regions && (allRegionsUnavailable || someRegionsUnavailable) ? (
+          <HelperText className={"pf-c-form__helper-text"}>
+            <HelperTextItem variant="warning" hasIcon>
```
Regarding my other comment, can we update this to be conditionally `warning` for `t("some_region_unavailable_helper_text")`
and
`error` for `t("all_region_unavailable_helper_text")}`

riccardo-forina commented:
Merging this as it is (should be good). We can track any further change as new PRs.
